Source/JavaScriptCore/dfg/DFGByteCodeParser.cpp

```diff
@@ -5270,31 +5270,45 @@ void ByteCodeParser::handleGetById(
         return;
     }
 
-    if (UNLIKELY(m_graph.compilation()))
-        m_graph.compilation()->noticeInlinedGetById();
-
     ASSERT(type == AccessType::GetById || type == AccessType::GetByIdDirect || !variant.callLinkStatus());
-    if (!variant.callLinkStatus() && variant.intrinsic() == NoIntrinsic) {
-        set(destination, loadedValue);
+
+    auto const getGetter = [&] {
+        if (JSValue getterValue = m_graph.tryGetConstantGetter(loadedValue))
+            return weakJSConstant(getterValue);
+
+        return addToGraph(GetGetter, loadedValue);
+    };
+
+    if (variant.intrinsic() != NoIntrinsic) {
+        auto const addChecks = [&] {
+            Node* getter = getGetter();
+            addToGraph(CheckIsConstant, OpInfo(m_graph.freeze(variant.intrinsicFunction())), getter);
+        };
+
+        if (handleIntrinsicGetter(destination, prediction, variant, base, addChecks)) {
+            if (UNLIKELY(m_graph.compilation()))
+                m_graph.compilation()->noticeInlinedGetById();
+            addToGraph(Phantom, base);
+            return;
+        }
+
+        // We couldn't handle this as an intrinsic--bail and emit a regular GetById
+        set(destination,
+            addToGraph(getById, OpInfo(identifier), OpInfo(prediction), base));
```

Comment:
This is not correct for 64bit. We should use getter call instead.

Reply:
For example, if it is not intrinsic getter, but if it has intrinsic, then, this change regresses it.
I think the right fix is using this only when we do not have callLinkInfo, or making non-failable in 32bit. Current change is regressing 64bit implementation, this is not OK.

Reply:
I see--when I looked before it seemed like it's not possible, in `GetByStatus::computeForStubInfoWithoutExitSiteFeedback` to get a `GetByVariant` with both a valid `intrinsic()` and missing a `callLinkStatus()`, but now I see it's possible to merge two variants and end up with one like this, right?

Is it enough to guard this fallback with `!variant.callLinkStatus()` ?

Reply:
> Is it enough to guard this fallback with !variant.callLinkStatus() ?

I've gone ahead and pushed a version that does this